### DISCUSSION TITLE
Devshell-files: Static files generators

### DISCUSSION THREAD

**Discussion by hugosenari** [numtide/devshell#152] [Show and tell] [answered]
Hi, everyone.

Firstly I would like to thank this project, it has almost all I've been looking for as developer :heart_eyes: 

For missing parts it makes easy to accomplish by myself, and since it may be helpful to others I'm sharing it here:

[Devshell-files](https://github.com/cruel-intentions/devshell-files) (sorry I'm not good at naming things) 

**What**: Helps static files creation - _text, json, yaml, toml and hcl_ - using _nix_ syntax
**Why**:  "_Config Sucks_", "_Configuration Complexity Curse_", "_Noyaml_", [etc](https://github.com/cruel-intentions/stylops#related)...

**Example**
```nix
# examples/hello.nix
{
  config.files.json."/hello.json".hello = "world";
  config.files.toml."/hello.toml".hello = "world";
  config.files.yaml."/hello.yaml".hello = "world";
  config.files.hcl."/hello.hcl".hello = "world";
  config.files.text."/hello.txt" = "world";
}
```

**Other features**: [gitignore templates](https://cruel-intentions.github.io/devshell-files/modules/gitignore.html), [license templates](https://cruel-intentions.github.io/devshell-files/modules/spdx.html), [modules2markdown](https://cruel-intentions.github.io/devshell-files/document.html) and [markdown2html](https://cruel-intentions.github.io/devshell-files/modules/mdbook.html)
**Projects using it**: [gh-action](https://github.com/cruel-intentions/gh-actions/) yaml file generator², [friendly modules](https://hugosenari.github.io/friendly-modules/index.html) random modules documentation
**How**: it is basically a copy of _home-manager_ [file-type](https://github.com/nix-community/home-manager/blob/master/modules/lib/file-type.nix) and use base git dir as _home_
**To-do**: ini files and ~asks if anything there would better be in devshell~

There are anything that would better be in devshell?

² to be honest isn't a project but result of 'how to' from [docs](https://cruel-intentions.github.io/devshell-files/module-spec.html#options) and it works so well that I'm using in other projects

**Comment by zimbatm** [comment, marked as answer]
Hi hugosenari,

Thanks for the kind words. With your module, devshell is coming closer to a home-manager on the project-level which is quite cool to see.

I need to take a closer look at the implementation but I wouldn't be opposed to adding some of those modules to the devshell extras. And/or list your project in the README.
